### PR TITLE
methods to manipulate atom types in ReactionManipulator

### PR DESCRIPTION
**Pull request by uli-f**
- added methods to perceive and unset the atom type configurations in ReactionManipulator
- fixed a javadoc errors wrt content in AtomContainerManipulator#percieveAtomTypesAndConfigureUnsetProperties

### CODE REVIEW

**Review by egonw — CHANGES_REQUESTED** [review]
Please add test methods for all new methods. And thanx for the extensive documentation!

File: base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java [hunk 1]
```diff
@@ -1541,15 +1545,15 @@ public static void percieveAtomTypesAndConfigureUnsetProperties(IAtomContainer c
 
     /**
      * This method will reset all atom configuration to UNSET.
-     *
+     * <br>
```
Comment (egonw) [inline base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java:1548]: In HTML terms, you should use `<p>` here, I think

File: base/standard/src/main/java/org/openscience/cdk/tools/manipulator/ReactionManipulator.java [hunk 2]
```diff
@@ -539,4 +529,64 @@ public static Set<IBond> findMappedBonds(IReaction reaction) {
         }
         return mapped;
     }
+
+    /**
+     * Convenience method to perceive atom types for all {@link IAtom IAtoms} of all components of the provided {@link IReaction}.
```
Comment (egonw) [inline base/standard/src/main/java/org/openscience/cdk/tools/manipulator/ReactionManipulator.java:534]: Is `{@link IAtom IAtoms}` what you meant to write?

Comment (uli-f) [reply base/standard/src/main/java/org/openscience/cdk/tools/manipulator/ReactionManipulator.java:534]: Yes, it is.
This links to IAtom but uses the text IAtoms for the link.

File: base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java (hunk 1 above)
Comment (uli-f) [reply base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java:1548]: I usually use both, depending on whether I want a paragraph or just a line break.

Happy to change to a paragraph if that is what you prefer.

**Comment by uli-f** [comment]
> Please add test methods for all new methods.

I decided against adding tests as the methods merely combine two (already tested) method calls. There is not logic in those methods other than a for loop.

I am happy to add tests if that is what you prefer (e.g., to achieve good test coverage numbers).
Just let me know and I am on it.

File: base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java (hunk 1 above)
Comment (egonw) [reply base/standard/src/main/java/org/openscience/cdk/tools/manipulator/AtomContainerManipulator.java:1548]: This is not a breaking issue for me :)

**Comment by egonw** [comment]
> I decided against adding tests as the methods merely combine two (already tested) method calls.

Understood. But... we write unit tests as much as now as for the future. The implementation is now simple, but in the future may not. I found it much more beneficial for maintainability if all assumption are explicit. It fixed a lot of code in the 2003-2008 period, even the silliest unit tests.

**Comment by uli-f** [comment]
> Understood. But... we write unit tests as much as now as for the future. The implementation is now simple, but in the future may not. I found it much more beneficial for maintainability if all assumption are explicit. It fixed a lot of code in the 2003-2008 period, even the silliest unit tests.

Okay, I see your point.
Will add (simple) tests now.

**Comment by uli-f** [comment]
Added tests now.
Looking forward to feedback 😃

**Review by egonw — APPROVED** [review]
Thanks for adding the tests.

**Comment by johnmay** [comment]
Looks fine, personally I don't think we use wildcard imports anywhere else mainly for convention. In the restricted case of ``org.openscience.cdk.interfaces.*`` I guess it is fine. It would be marginally more efficient to loop the component types rather than use the less efficient ``getAllMolecules``. The fastest toolkits represent reactions as molecules anyways and so technically it's already possible to do it like this:

```
java
IReaction rxn = ...;
IAtomContainer mol = ReactionManipulator.toMolecule(rxn);
AtomContainerManipulator.percieveAtomTypesAndConfigureAtoms(mol);
rxn = ReactionManipulator.toReaction(mol);
```

Although that would be pretty silly :-).

**Pull request merged**